dev_guide/builds.adoc

```diff
@@ -519,9 +531,42 @@ For example, defining a custom HTTP proxy to be used during build:
 ----
 ====
 
+[[source-code]]
+
+== Git Repository Source Options
```

Comment:
(we could do the same for the Strategies... adding a Build Strategies section and putting s2i/docker/custom as subsections under that)